makeJsDist.sh

```diff
@@ -9,6 +9,12 @@ fi
 pushd gui-js
 npm run export:package:linux
 pushd dist/executables/
-mv linux-unpacked $name
-rm $name/resources/node-addons/*.node
+mkdir $name
+mv linux-unpacked $name/app
```

Comment:
Could you please enlighten me why this directory needs to sit under a subdirectory?

Also - are there requisite changes required in the minsky.spec and debian.rules files to get the packaging to work.

Reply:
> Could you please enlighten me why this directory needs to sit under a subdirectory?

To separate the electron bundle from the desktop and icon file for integration in the system. It eases the way packages are built too.

> Also - are there requisite changes required in the minsky.spec and debian.rules files to get the packaging to work.

Here would be the relevant sections in `minsky.spec`:

```
%install
mkdir -p %{buildroot}/usr/lib/minsky %{buildroot}/%{_bindir} %{buildroot}/%{_datadir}/icons/hicolor/scalable/apps %{buildroot}/%{_datadir}/applications
cp -r app/* %{buildroot}/usr/lib/minsky
ln -sf /usr/lib/minsky/minsky %{buildroot}/%{_bindir}
cp minsky.svg %{buildroot}/%{_datadir}/icons/hicolor/scalable/apps
cp minsky.desktop %{buildroot}/%{_datadir}/applications

...

%files
%defattr (-,root,root)
%{_bindir}/minsky
%dir /usr/lib/minsky
/usr/lib/minsky/*
%{_datadir}/icons/hicolor/scalable/apps/minsky.svg
%{_datadir}/applications/minsky.desktop
```

And `debian.rules`:

```
# Add here commands to install the package
	mkdir -p $(DEST_DIR)/usr/lib/minsky $(DEST_DIR)/usr/bin $(DEST_DIR)/usr/share/icons/hicolor/scalable/apps $(DEST_DIR)/usr/share/applications
	echo cp -r app/* $(DEST_DIR)/usr/lib/minsky
	cp -r app/* $(DEST_DIR)/usr/lib/minsky
	ln -sf /usr/lib/minsky/minsky $(DEST_DIR)/usr/bin
	echo cp -r minsky.svg $(DEST_DIR)/usr/share/icons/hicolor/scalable/apps
	cp -r minsky.svg $(DEST_DIR)/usr/share/icons/hicolor/scalable/apps
	echo cp -r minsky.desktop $(DEST_DIR)/usr/share/applications
	cp -r minsky.desktop $(DEST_DIR)/usr/share/applications
# --- end custom part for installing
```

Reply:
Thanks - I'll be sure to update the files when I do the next release.

I fixed all of the out of bounds accesses causing problems since merging your last PR, so this should be good to go.